### DISCUSSION TITLE
Monthly invoice send by cronjob ?

### DISCUSSION THREAD

**Discussion by cube4d** [symfony/symfony#45820] [Q&A] [answered]
Hi, 

i am really new to all, but i have build my first small symfony webapp, I have my database with all my costumers and now i have my controller wich will go through all costumers and send an email with invoices. (I want to do it every first day of the month )

My question is: 

how is something like that done in the real world ? :-)
I thougt i can run my controller monthly with a cronjob from the server, but a cronjob is no authenticated user :-) and i dont want to 
make the controller acessible by public. 

I am just happy with a few ideas thank you.

**Comment by stof** [comment, marked as answer]
make the cronjob run a command instead of making it perform an HTTP request to a controller

**Reply by cube4d** [reply]
Thank you for your answer, then you mean like so:

php /xxxx.com/httpdocs/xxxxx/src/Controller/InvoiceController.php

**Reply by stof** [reply]
No. I mean using a CLI command instead of a controller: https://symfony.com/doc/current/console.html

**Reply by cube4d** [reply]
Ok thanks i will read a little bit

**Reply by cube4d** [reply]
Hey thank you sooo much... it  worked :-)
